sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java

```diff
@@ -279,99 +321,432 @@ public EventData addContext(String key, Object value) {
+     */
+    public byte[] getUserId() {
+        return amqpAnnotatedMessage.getProperties().getUserId();
+    }
+
+    /**
+     * Sets the message id.
+     *
+     * @param userId The message id to be set.
+     *
+     * @return The updated {@link EventData}.
+     * @throws IllegalArgumentException if {@code messageId} is too long.
+     */
+    public EventData setUserId(byte[] userId) {
+        amqpAnnotatedMessage.getProperties().setUserId(userId);
+        return this;
+    }
+
+    /**
+     * Gets the message id.
+     *
+     * <p>
+     * The message identifier is an application-defined value that uniquely identifies the message and its payload. The
+     * identifier is a free-form string and can reflect a GUID or an identifier derived from the application context.
+     * </p>
+     *
+     * @return Id of the {@link EventData}.
+     */
+    public String getMessageId() {
+        String messageId = null;
+        AmqpMessageId amqpMessageId = amqpAnnotatedMessage.getProperties().getMessageId();
+        if (amqpMessageId != null) {
+            messageId = amqpMessageId.toString();
+        }
+        return messageId;
+    }
+
+    /**
+     * Sets the message id.
+     *
+     * @param messageId The message id to be set.
+     *
+     * @return The updated {@link EventData}.
+     * @throws IllegalArgumentException if {@code messageId} is too long.
+     */
+    public EventData setMessageId(String messageId) {
+        checkIdLength("messageId", messageId, MAX_MESSAGE_ID_LENGTH);
+        AmqpMessageId id = null;
+        if (messageId != null) {
+            id = new AmqpMessageId(messageId);
+        }
+        amqpAnnotatedMessage.getProperties().setMessageId(id);
+        return this;
+    }
+
+    /**
+     * Gets the subject for the message.
+     *
+     * <p>
+     * This property enables the application to indicate the purpose of the message to the receiver in a standardized
+     * fashion, similar to an email subject line. The mapped AMQP property is "subject".
+     * </p>
+     *
+     * @return The subject for the message.
+     */
+    public String getSubject() {
+        return amqpAnnotatedMessage.getProperties().getSubject();
+    }
+
+    /**
+     * Sets the subject for the message.
+     *
+     * @param subject The application specific subject.
+     *
+     * @return The updated {@link EventData} object.
+     */
+    public EventData setSubject(String subject) {
+        amqpAnnotatedMessage.getProperties().setSubject(subject);
+        return this;
+    }
+
+    /**
+     * Gets the "to" address.
+     *
+     * <p>
+     * This property is reserved for future use in routing scenarios and presently ignored by the broker itself.
+     * Applications can use this value in rule-driven
+     * auto-forward scenarios to indicate the intended logical destination of the message.
+     * </p>
+     *
+     * @return "To" property value of this message
+     */
+    public String getTo() {
```

Comment:
EH might not need this.

Reply:
It does not.

Reply:
According to your comment, remove unnecessary parameters, variables and methods in the new version